allennlp/tests/models/language_model_test.py

```diff
@@ -0,0 +1,156 @@
+# pylint: disable=invalid-name,arguments-differ,abstract-method
+import numpy as np
+import pytest
+
+from allennlp.common.testing import ModelTestCase
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.params import Params
+from allennlp.models import Model
+
+class TestUnidirectionalLanguageModel(ModelTestCase):
```

Comment:
Having an extra option for the directionality has made it difficult to see at a glance that the different tests are behaving as expected. I didn't do a great job with this when I added the transformer test, but in retrospect it should be pretty simple.

Roughly what needs to happen is a) set `expected_embedding_shape` in the constructor and b) check against this later. Similarly for `expected_backward_loss` and `breaking_contextualizer_direction`. Actually, I think the inherited test methods will be used too. (Worth verifying that works as expected.) So a lot of code can be removed.

As an aside, there is a balance to be struck between straightforward tests with modest duplication and parameterized ones. Here I'd lean towards parameterizing.